### ISSUE TITLE
Allow integrated terminal to run as login shell

### ISSUE THREAD

**Issue by Luchillo** [Microsoft/vscode#7263]
- VSCode Version: 1.2.0
- OS Version: Ubuntu 16.04

While testing the integrated terminal i realized that my native terminal aliases and custom commands are missing, say i have an alias like `alias ll='ls -FGlahp'` defined in my `.bash_profile` which gives me colored folder list with `/` terminated names for folders, if i try the same in the integrated terminal the output is the following, is it possible to embed the native terminal or config in the integrated one the import to the same config files of the native one?

```sh
ll: command not found
```

Steps to Reproduce:

1. In Ubuntu just put a custom alias in the config files of your bash session, most likely `.bash_profile` or `.bashrc`.
2. Try to use that new custom alias in the VS Code integrated terminal.

**Comment by Luchillo** [comment]
I've tried to call `source ~/.bash_profile` and it worked perfectly, maybe if we could configure the source file for the integrated terminal session this would be a finished feature.

**Comment by Luchillo** [comment]
Ok i've been testing and the difference with native and integrated is the fact that i configured the native to `Run as login shell`, idk if the integrated one has the config option like the native like this in profile:
![image](https://cloud.githubusercontent.com/assets/9698639/15835511/f819f0da-2bf5-11e6-9dd1-c986cc4828ef.png)

**Comment by gertcuykens** [comment]
+1 for startup command feature so you can launch `source ~/.bash_profile` automaticlly

**Comment by Tyriar** [comment]
Yes currently the terminal runs as a non-login shell, I'll add this to the backlog.

In the meantime, if you don't want differences between login and non-login shells, I recommend moving the .bash_profile contents to your .bashrc and adding the following to your .bash_profile:


```
# .bash_profile
if [ -r ~/.bashrc ]; then
  . ~/.bashrc
fi
```

**Comment by Luchillo** [comment]
I already have the source to `.bashrc` in `.bash_profile` however i will avoid editing the provided `.bashrc` by the system, i think i will stick to the native one until we can configure the launch of the integrated terminal.

**Comment by Tyriar** [comment]
@Luchillo it should be safe to just add their own stuff to the bottom of the default `.bashrc` file.

**Comment by coreh** [comment]
Workaround: 

1) Create a file named `/usr/local/bin/bash-login` (or whatever you'd like to name it) and add the following lines to it (You might need root to create that file depending on your setup):

```bash
#!/bin/bash
bash -l
```

2) Make it executable with `chmod +x /usr/local/bin/bash-login`.

3) On your VSCode user settings, set:

```JSON
{
  "terminal.integrated.shell.linux": "/usr/local/bin/bash-login"
}
```

4) Profit ✨ 

![](https://cldup.com/6lQPeaCwcJ.png)

**Comment by Tyriar** [comment]
@coreh if adding the `-l` arg is all there is to it then this should make it into the next release in the form of https://github.com/Microsoft/vscode/issues/7266

**Comment by Luchillo** [comment]
@Tyriar Yes, `-l` flag, that's all it needs.
Also i won't edit the `.bashrc` but put all changes in `.bash_profile` in order to have compatibility with OSX systems, i have my `.bash_profile` in a repo that saves the changes so that i get similar behaviour in both OSX and Ubuntu, same aliases and OS dependent commands.

**Comment by Luchillo** [comment]
@coreh Awesome workaround, how didn't i tough about it?

**Comment by Tyriar** [comment]
Going to close this in favor of https://github.com/Microsoft/vscode/issues/7266 which will allow you to do this in 1.3.0:

```json
{
  "terminal.integrated.shell.linux": "bash",
  "terminal.integrated.shellArgs.linux": [ "-l" ]
}
```

In the meantime if you want this working on stable you can do the workarounds that myself https://github.com/Microsoft/vscode/issues/7263#issuecomment-224140256 or @coreh https://github.com/Microsoft/vscode/issues/7263#issuecomment-224207986 suiggested.

**Comment by Luchillo** [comment]
Cool.

**Comment by tiagolo** [comment]
guys, what about in windows?? 
Will I be able to do the same with Windows 10 Ubuntu Bash, or Git-Bash? 
Anyone can give a tip on creating a **.bat** file to launch bash as a login shell?

**Comment by Tyriar** [comment]
@tiagolo not sure it's even a concept on git bash, for Ubuntu Bash it's all about whether the executable accepts command line arguments, not really a question for us.

**Comment by tiagolo** [comment]
@Tyriar actualy the Git-Bash shell has the same issue with the non-login shell, so after some search I could get any with **batch file** like this:

```Batch
@ECHO OFF
"C:\Program Files\Git\bin\bash.exe" --login -i
```

**Comment by kattak2k** [comment]
Just to help others, to fix the issue in Ubuntu 17.04... try below code

```
{
    "terminal.integrated.shellArgs.linux": ["--login"]
}

```
result: basically it formats to  **/bin/bash --login**

**Comment by w5m** [comment]
Overriding default settings with the following user settings worked nicely for me on Windows...
```
{ 
   // Git Bash
    "terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",
    // start a login shell to pick up aliases from .bash_profile
    "terminal.integrated.shellArgs.windows": ["--login"]
}
```
